manticore/core/state.py

```diff
@@ -143,16 +142,37 @@ def setstate(state, value):
             raise TerminateState(e.message, testcase=True)
 
         #Remove when code gets stable?
-        assert self.platform.constraints is self.constraints
-        assert self.mem.constraints is self.constraints
+        assert self._platform._constraints is self._constraints
+        assert self.mem._constraints is self._constraints
         return result
 
+    @property
+    def input_symbols(self):
+        return self._input_symbols
+
+    @property
+    def context(self):
+        return self._context
+
+    @property
+    def platform(self):
+        return self._platform
+
+    @property
+    def constraints(self):
```

Comment:
Do we want to return `_platform.constraints` here and remove the cached `_constraints_ reference altogether?